feat!: remove transcript file field

samuelallan72 opened the pull request:
## Description

The transcript file field is problematic, as it stores the file in a location that is not included in course exports. This means if a transcript file is uploaded, then the course exported and imported into another Open edX instance, the file is lost.

The field is also not currently in use as far as we're aware, so we can simply remove the field.

This still allows timed transcripts, via the transcript url field.

In future if we need support, we will need to consider a better way to handle file uploads. For example, the core Video XBlock handles handout file uploads by directly uploading them to the course Files assets. This isn't ideal though. There are also other methods being considered for future development, so we may have a better supported method in the future.

## Supporting information

Private-ref: https://tasks.opencraft.com/browse/BB-10249

## Testing instructions

- deploy the previous revision of this xblock (wgu-release branch) to an instance
- create a component with this xblock, uploading a transcript file
- update the xblock to this revision
- navigate to the previously created component, and verify the component still works, albeit with the transcript missing. (This is to test that it gracefully upgrades; we don't want it to throw an error.)
- edit the xblock again, and add a transcript via the url field (please let me know if you find a good sample webvtt hosted online)
- check the edit view and student view and verify the text and fields, especially around transcripts, makes sense still after the removal of the file field
- Verify the timed transcript still works as expected when using the transcript url field.

samuelallan72 committed:
feat!: remove transcript file field

The transcript file field is problematic,
as it stores the file in a location
that is not included in course exports.
This means if a transcript file is uploaded,
then the course exported and imported into another Open edX instance,
the file is lost.

The field is also not currently in use as far as we're aware,
so we can simply remove the field.

This still allows timed transcripts, via the transcript url field.

In future if we need support, we will need to consider a better way to handle file uploads.
For example, the core Video XBlock handles handout file uploads
by directly uploading them to the course Files assets.
This isn't ideal though.
There are also other methods being considered for future development,
so we may have a better supported method in the future.

Private-ref: https://tasks.opencraft.com/browse/BB-10249

farhaanbukhsh reviewed (changes requested):
@samuelallan72 Great work here it's always good to see code being removed, I think you might have missed two lines [here](https://github.com/open-craft/xblock-audio/blob/wgu-release/audio/public/js/audio_edit.js#L45).

also I have used these links to test:

- https://ableplayer.github.io/ableplayer/media/frenchsong.mp3
- https://ableplayer.github.io/ableplayer/media/frenchsong_chapters_en.vtt

samuelallan72 commented:
Thanks @farhaanbukhsh , and thanks for the example urls. :)

I've addressed your comment, and I also discovered and removed the reference to uploaded files in the readme.

Ready for another review pass. :)

farhaanbukhsh reviewed (approved):
👍

✅ I tested this on Tutor devstack that.
- The audio block is working properly
- Rendering the audio and playing it
- The transcript is working properly
- Necessary code has been removed.
✅ I read through the code
✅   I checked for accessibility issues
✅   Includes documentation

farhaanbukhsh commented:
@samuelallan72 Please squash your commits before merging. :) thank you for the work here.

samuelallan72 commented:
Done, thanks @farhaanbukhsh :)
